File: Autos-stock-manager-services/src/main/java/org/autostock/dtos/VenteCreateDto.java
package org.autostock.dtos;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
public class VenteCreateDto {
    private Long idVoiture;
    private Long idClient;
    private Long idVendeur;
    private BigDecimal prixFinal;
    private String modePaiement; // "CASH", "CARD", etc. (MethodePaiement.value)
    // Date saisie dans le formulaire ('YYYY-MM-DD'). Absente du DTO, elle etait
    // ignoree par Jackson et la vente etait systematiquement datee du jour.
    private LocalDate dateVente;

    // getters / setters
}

File: autos-stock-client/src/app/models/VenteCreateDto.ts
import {MethodePaiement} from "./enums/MethodePaiement";

export interface VenteCreateDto {
  idVoiture: number;
  idClient: number;
  idVendeur: number;
  dateVente: string;       // ISO (ex: 2025-11-26T15:00:00Z) ou 'YYYY-MM-DD'
  prixFinal: number;
  modePaiement: MethodePaiement;
  acompteMontant?: number; // optionnel, pour un paiement initial
}

export interface VenteDto extends VenteCreateDto {
  id: number;
}
